introspection-engine/connectors/mongodb-introspection-connector/src/sampler/statistics.rs

```diff
@@ -105,7 +105,13 @@ impl<'a> Statistics<'a> {
     /// - if model is foo and field is bar, the type is FooBar
     /// - if a model already exists with the name, we'll use FooBar_
     fn composite_type_name(&self, model: &str, field: &str) -> Name {
-        let name = Name::Model(format!("{}_{}", model, field).to_case(Case::Pascal));
+        let combined = format!("{}_{}", model, field);
+
+        let name = if combined.chars().count() == combined.len() {
```

Comment:
I get the logic and it makes sense. Nitpick but alternatively `combined.chars().all(|c| c.is_ascii())` would do the same and is possibly a little bit more readable.